pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/XPathPanelController.java

```diff
@@ -112,9 +120,58 @@ public void initialize(URL location, ResourceBundle resources) {
                            // Reevaluate XPath anytime the expression or the XPath version changes
                            .or(xpathVersionProperty().changes())
                            .subscribe(tick -> parent.refreshXPathResults());
+
+        xpathExpressionArea.plainTextChanges()
+                           .map(m -> {
+                               Integer a = xpathExpressionArea.getText().lastIndexOf("/", m.getInsertionEnd());
+                               String s = xpathExpressionArea.getText().substring(a + 1);
+                               return Tuples.t(a, s);
+                           })
+                           .filter(t -> t._2.matches("[a-zA-Z]+"))
+                           .subscribe(e -> autoComplete(e._1, e._2));
+    }
+
+    private void autoComplete(int slashPosition, String input) {
+        System.out.println(input);
+        autoCompletePopup.getItems().clear();
+
+        XPathSuggestions xPathSuggestions = new XPathSuggestions(parent.getLanguageVersion().getLanguage());
+        List<String> suggestions = xPathSuggestions.getXPathSuggestions(input.trim());
+
+        List<MenuItem> resultToDisplay = suggestions.stream().map(m -> new MenuItem(m)).limit(5).collect(Collectors
+                                                                                                            .toList());
+
+        //TODO: Work on the implementation of the Result to be selected and added to the Code Area
+        autoCompletePopup.getItems().addAll(resultToDisplay);
+
+        if (xpathExpressionArea.getText().length() > 0) {
+
+            xpathExpressionArea.addEventHandler(KeyEvent.KEY_PRESSED, t -> {
```

Comment:
This event handler is added every time the suggestions are updated. You should only set it only once in the initialise method. The only thing that changes then is the menu items.